### PR TITLE
improved DX and api tweaks

### PR DESCRIPTION
**Pull request by gabriel-andreescu**
Changed comments/changelogs endpoints to take an assetid,
 if it's not provided the latest 100 are shown as discussed in https://github.com/anegostudios/vsmoddb/pull/9

Added files/ in front of release mainfile for consistency,
 this is how the logo url is also returned

Added mod screenshots and trending points to the mod endpoint

Added lastreleased/trending points to the mods endpoint

Added ability to order the mods endpoint by limited list of columns
 and decide the direction

Improved development experience with some sample data and a way to setup a dev env without vscode, also fixed some annoyances encountered during my testing, they were documented with comments.

Updated README.md to reflect the changes

### CODE REVIEW

File: api.php [hunk 1]
```diff
@@ -164,28 +184,53 @@ function listMod($modid)
...
+
 	$mod = array(
 		"modid" => intval($row["modid"]),
 		"assetid" => intval($row["assetid"]),
 		"name" => $row['name'],
 		"text" => $row['text'],
 		"author" => $row['author'],
 		"urlalias" => $row['urlalias'],
-		"logofilename" => $row['logofilename'] ? "asset/{$row['assetid']}/" . $row['logofilename'] : null,
+		"logo" => $row['logofilename'] ? "files/asset/{$row['assetid']}/" . $row['logofilename'] : null,
```
Comment (tyronx) [inline api.php:217]: the `files/` prefix is not necessary - by definition all file paths need to use the base url by convention as documented here: https://github.com/anegostudios/vsmoddb#urls

File: db/tables.sql [hunk 2]
```diff
@@ -386,3 +386,60 @@ INSERT INTO `moddb`.`role` (`roleid`, `code`, `name`, `created`, `sortorder`, `l
...
+INSERT INTO `moddb`.`asset` (createdbyuserid, editedbyuserid, statusid, assettypeid, code, name, text, tagscached, created, lastmodified, numsaved) VALUES (1,1,2,1,NULL,'Example Mod 10', '<p style="text-align: center;">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque leo sem, ultrices vel enim vel, pretium fringilla nisi. Nunc ac massa hendrerit, semper est sed, blandit eros. Sed at placerat lorem, viverra lacinia nibh. Mauris eu nunc a augue rhoncus pharetra ac eu nulla. Fusce elementum sapien sit amet sapien pellentesque, eget porttitor quam eleifend. Maecenas imperdiet justo dolor, id bibendum purus ornare vel. Morbi commodo porttitor nisi, sed finibus eros blandit eget. Nulla quis rhoncus urna.</p>', '', CURRENT_TIMESTAMP, CURRENT_TIMESTAMP, 1);
+
+COMMIT;
```
Comment (tyronx) [inline db/tables.sql:445]: Example data please into a separate file, e.g. sampledata.sql

File: api.php (hunk 1 above)
Comment (tyronx) [reply api.php:217]: alternatively we change the convetion in the docs

Comment (gabriel-andreescu) [reply api.php:217]: I've missed the convention, maybe removing the `files/` prefix from `/api/mods` endpoint is a better change

**Comment by tyronx** [comment]
thank you for this!

**Comment by tyronx** [comment]
wait, removing `/files` everywhere will break backwards compatibility with 3rd party tools using the api

**Comment by tyronx** [comment]
new api funtionality is now online

**Comment by gabriel-andreescu** [comment]
> wait, removing `/files` everywhere will break backwards compatibility with 3rd party tools using the api. i'll have to revert that.

if it is ok with you, when I get some time I'd like to implement a simple versioning system (nothing fancy, just url based) so changes like this can happen

**Comment by tyronx** [comment]
sure, feel free

**Pull request merged**